pandas/core/dtypes/cast.py

```diff
@@ -647,7 +647,14 @@ def conv(r, dtype):
 
 def astype_nansafe(arr, dtype, copy=True):
     """ return a view if copy is False, but
-        need to be very careful as the result shape could change! """
+        need to be very careful as the result shape could change!
+
+    Parameters
+    ----------
+    arr : ndarray
+    dtype : np.dtype
+    copy : bool, default True
```

Comment:
Can you specify here what happens with False? (will always do view (except for object), even if the itemsize is incompatible)